src/engraving/libmscore/note.cpp

```diff
@@ -2379,19 +2384,15 @@ void Note::layout2()
             }
         }
         // if TAB and stems through staff
-        if (staff()->isTabStaff(chord()->tick())) {
-            const Staff* st = staff();
-            const StaffType* tab = st->staffTypeForElement(this);
-            if (tab->stemThrough()) {
-                // with TAB's, dot Y is not calculated during layoutChords3(),
-                // as layoutChords3() is not even called for TAB's;
-                // setDotY() actually also manages creation/deletion of NoteDot's
-                setDotY(DirectionV::AUTO);
-
-                // use TAB default note-to-dot spacing
-                dd = STAFFTYPE_TAB_DEFAULTDOTDIST_X * spatium();
-                d = dd * 0.5;
-            }
+        if (staff()->isTabStaff(chord()->tick()) && tab->stemThrough()) {
```

Comment:
This staff()->isTabStaff(chord()->tick()) is already called at least twice in one function

Reply:
no longer!